[Ubuntu] Adding Bicep CLI

MarcusFelling opened the pull request:
# Description
PR with bicep cli installation for Ubuntu images. Includes changes from #3600 for Windows.

#### Related issue:
#3461 

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

miketimofeev commented:
@MarcusFelling please remove the windows changes

MarcusFelling commented:
> @MarcusFelling Marcus Felling FTE please remove the windows changes

@miketimofeev Cool if I just wait for that PR to be completed?

miketimofeev commented:
@MarcusFelling I guess there will be conflicts then. It's better to create different branch for ubuntu

MarcusFelling commented:
> @MarcusFelling Marcus Felling FTE I guess there will be conflicts then. It's better to create a different branch for ubuntu

@miketimofeev There should be conflicts, but I rebased and dropped Windows commits. Take a look and let me know if you need anything else.

miketimofeev commented:
@MarcusFelling great, thanks! I'll push the builds

miketimofeev commented on images/linux/scripts/installers/bicep.sh:
```diff
@@ -0,0 +1,14 @@
+################################################################################
+
+# Fetch the latest Bicep CLI binary
+curl -Lo bicep.bin https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64
```
I suggest you use the download_with_retries function here https://github.com/actions/virtual-environments/blob/main/images/linux/scripts/helpers/install.sh#L7
Something like 
`download_with_retries $URL "." "bicep"`

MarcusFelling replied:
Script has been updated to use download_with_retries from helper.

miketimofeev commented on images/linux/scripts/installers/bicep.sh:
```diff
@@ -0,0 +1,14 @@
+# Mark it as executable
+chmod +x ./bicep.bin
+# Add bicep to PATH (requires admin)
+sudo mv ./bicep.bin /usr/local/bin/bicep
```
why do we need .bin extension prior to moving the file?

MarcusFelling replied:
No reason it couldn't change. I wanted to stay in line with our official install script: https://github.com/Azure/bicep/blob/main/docs/installing.md#linux

AlenaSviridenko commented on images/linux/ubuntu1604.json:
```diff
@@ -194,6 +194,7 @@
                 "{{template_dir}}/scripts/installers/azure-cli.sh",
                 "{{template_dir}}/scripts/installers/azure-devops-cli.sh",
                 "{{template_dir}}/scripts/installers/basic.sh",
+                "{{template_dir}}/scripts/installers/bicep.sh",
```
Please, also remove changes for Ubuntu 16.04. It is currently under deprecation process, we do not update it anymore, and will remove on September, 20. https://github.com/actions/virtual-environments/issues/3287

al-cheb commented on images/linux/scripts/installers/bicep.sh:
```diff
@@ -0,0 +1,16 @@
+source $HELPER_SCRIPTS/install.sh
+
+# Install Bicep CLI
+download_with_retries "https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64" "." "bicep.bin"
```
nitpick: use dst param and remove mv statement
```
download_with_retries "https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64" "/usr/local/bin" "bicep"
chmod +x /usr/local/bin/bicep
```

MarcusFelling replied:
I'd prefer to stay in line with our official install steps here: https://github.com/Azure/bicep/blob/main/docs/installing.md#linux

al-cheb commented on images/linux/scripts/SoftwareReport/SoftwareReport.Tools.psm1:
```diff
@@ -25,6 +25,11 @@ function Get-BazeliskVersion {
     $bazeliskVersion = $result.Output | Select-String "Bazelisk version:" | Take-OutputPart -Part 2 | Take-OutputPart -Part 0 -Delimiter "v"
     return "Bazelisk $bazeliskVersion"
 }
+function Get-BicepVersion {
```
nitpick: please add a new line above `function Get-BicepVersion`

AlenaSviridenko commented:
/azp run ubuntu1804, ubuntu2004
